tests/modules/test_text_processor.py

```diff
@@ -260,6 +274,139 @@ def test_subscribe_twice_hindi(self, texting_mock, logging_info_mock, logging_er
     @patch("logging.error")
     @patch("logging.info")
     @patch("modules.text_processor.Texter.send")
+    def test_subscribe_twice_doesnt_change_preg_update_english(self, texting_mock, logging_info_mock, logging_error_mock):
+        t = TextProcessor(phone_number="1-111-1114")
+        self.assertFalse(Contact.objects.filter(name="Rose", phone_number="1-111-1114").exists())
+        first_response = t.process("JOIN ROSE 25-11-2012")
+        self.assertEqual(first_response, msg_subscribe("English").format(name="Rose"))
+        texting_mock.assert_called_with(message=first_response, phone_number="1-111-1114")
+        self.assertTrue(Contact.objects.filter(name="Rose", phone_number="1-111-1114", preg_update=False).exists())
+        self.assertTrue(t.get_contacts().exists())
+        t2 = TextProcessor(phone_number="1-111-1114")
+        second_response = t2.process("JOIN ROSE 25-11-2012")
+        self.assertTrue(t2.get_contacts().exists())
+        self.assertEqual(second_response, msg_already_sub("English"))
+        texting_mock.assert_called_with(message=second_response, phone_number="1-111-1114")
+        logging_error_mock.assert_called_with("Contact for Rose at 1-111-1114 was subscribed but already exists!")
+        contacts = Contact.objects.filter(name="Rose", phone_number="1-111-1114", preg_update=False)
+        self.assertEqual(contacts.count(), 1)
+        self.assertTrue(t2.get_contacts().exists())
+
+    @patch("logging.error")
+    @patch("logging.info")
+    @patch("modules.text_processor.Texter.send")
+    def test_subscribe_twice_doesnt_change_preg_update_hindi(self, texting_mock, logging_info_mock, logging_error_mock):
+        t = TextProcessor(phone_number="1-111-1115")
+        first_response = t.process(hindi_remind() + " SANJIV 25-11-2012")
+        self.assertEqual(first_response, msg_subscribe("Hindi").format(name="Sanjiv"))
+        texting_mock.assert_called_with(message=first_response, phone_number="1-111-1115")
+        self.assertTrue(Contact.objects.filter(name="Sanjiv", phone_number="1-111-1115", preg_update=False).exists())
+        self.assertTrue(t.get_contacts().exists())
+        t2 = TextProcessor(phone_number="1-111-1115")
+        self.assertTrue(t2.get_contacts().exists())
+        second_response = t2.process(hindi_remind() + " SANJIV 25-11-2012")
+        self.assertEqual(second_response, msg_already_sub("Hindi"))
+        texting_mock.assert_called_with(message=second_response, phone_number="1-111-1115")
+        logging_error_mock.assert_called_with("Contact for Sanjiv at 1-111-1115 was subscribed but already exists!")
+        contacts = Contact.objects.filter(name="Sanjiv", phone_number="1-111-1115", preg_update=False)
+        self.assertTrue(contacts.exists())
+        self.assertEqual(contacts.count(), 1)
+        self.assertTrue(t2.get_contacts().exists())
```

Comment:
This test doesn't actually seem to test anything about the preg update.

Reply:
See my answer for the English test above.